### ISSUE TITLE
POI: ikony pásového značení u rozcestníků

### ISSUE THREAD

**Issue by vop-cz** [osmcz/osmcz#122]
Ke zvážení malá featura:
U otevřeného detailu POI v případě že jde o rozcestník navrhuji v prvním řádku vpravo vedle názvu rozcestníku generovat grafické značky tras, které jím procházejí (z hodnot osmc:symbol=\* v jeho příslušných relaci/ích).

**Comment by mkyral** [comment]
Zdar,
to by asi šlo. API query je: http://www.openstreetmap.org/api/0.6/node/3866018063/relations

Nechtěl by jsi připravit ty obrázky? Abych se s tím moc nezdržoval ;-)

**Comment by vop-cz** [comment]
obrázky: stačí (přinejmenším pro začátek) odsud?
http://wiki.openstreetmap.org/wiki/WikiProject_Czech_Republic/Editing_Standards_and_Conventions#Turistick.C3.A9_zna.C4.8Den.C3.AD

**Comment by mkyral** [comment]
No tam nejsou všechny kombinace barev ;-)

Koukám, že jako základ by se dalo použít http://wiki.openstreetmap.org/wiki/WikiProject_Czech_Republic/OTM_zna%C4%8Dkov%C3%BD_kl%C3%AD%C4%8D

Je to tam v svg. Jen by bylo potřeba dodělat ty čtyři základní barvy.

**Comment by zbycz** [comment]
nebo by šlo obsah SVG normálně nacpat do HTML kódu stránky a změnit barvu. Když jsem ho optimalizoval, je docela jednoduché:

``` svg
<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" version="1"><path fill="#fff" stroke="#000" stroke-width=".156" d="M.078 11.28h15.844v4.643H.078zm0-11.2h15.844v4.643H.078z"/><path fill="red" stroke="#000" stroke-width=".156" d="M.078 5.68h15.844v4.643H.078z"/></svg>
```

Taky by šlo dynamicky vykreslovat jednu ikonku s více barvama jako to dělá KČT na stromy, ale pro začátek klidně stačí tyto ikonky. Vlastně by se mi docela líbilo, když by u POI bylo dole zobrazeno součastí jakých relací je a tohle SVG by bylo úplně klasická ikonka "objektu relace".

Podobně to má [osm.org/node/370617054](http://osm.org/node/370617054)

**Comment by mkyral** [comment]
> nebo by šlo obsah SVG normálně nacpat do HTML kódu stránky a změnit barvu. Když jsem ho optimalizoval, je docela jednoduché:

No spíše jsem uvažoval nad svg spritem, či jak se to jmenuje. Ony do toho totiž patří i různé odbočky a přijde mi lepší mít jedno velké svg než hromadu html kódu.

> Vlastně by se mi docela líbilo, když by u POI bylo dole zobrazeno součastí jakých relací je a tohle SVG by bylo úplně klasická ikonka "objektu relace".

Pro mappery by to bylo jistě zajímavé, ale normálnímu uživateli to moc neřekne. Tam by se to muselo nějak šikovně obalit nějakou lehce srozumitelnou omáčkou.

Taky by bylo dobré si nejprve definovat, kde relace chceme  zobrazovat. Zatím bych zůstal u těch rozcestníků, později bych přidal zastávky MHD a pak třeba i další typy POI.

**Comment by vop-cz** [comment]
Jeste to zkomplikuju a pridam dalsi uzitnou vlastnost - pokud se mi podari jednoduse automatizovane zjistit, ze je to guidepost na zacatku te relace, pak by ta znacka mela odpovidat (tedy ctverecek v bilem poli)...

**Comment by zbycz** [comment]
Čtvereček v bílém poli ani neznám, to by mě dost zmátlo :-))

Jinak nejlíp by bylo využívat rovnou osmc:symbol, který stejně u relace existuje. Řekl bych že někde (hikebikemap.org) musí existovat nějaký napsaný program, který ze zadaného tagu osmc:symbol udělá SVGčko.. to by bylo navíc světově univerzální.

**Comment by vop-cz** [comment]
> Jinak nejlíp by bylo využívat rovnou osmc:symbol, který stejně u relace existuje.

Ano, to jsem původně myslel. Naráží to minimálně na dvě "výjimky", když vycházím z toho, co má vlastně celá ta featura uživateli přinést (a asi by to měla být informace, ke které relaci se ten POI vztahuje):
1) cesta, kde není fyzické liniové značení a dá se očekávat, jak by vypadalo, např. naučné stezky, u kterých (minimálně část mapperů) prosazuje logickou vlastnost "není-li fyzicky značka v terénu, neoznačuje se osmc:symbol"... pak by musela logika vycházet jen z tagů (což možná není problém, nutno zjistit v praxi...)
2) cesta, kde není fyzické liniové značení a NEDÁ se očekávat, jak by vypadalo, protože to jsou klasické "bílé" šipky, např. k blízkému nádraží apod.

Trochu to celé navazuje na připravovaný návrh na zjednodušení tagování a sjednocení postupů při netypických situacích... Ten návrh existuje a Tom chce, abych ho pustil ven do diskuse via Weekly...

**Comment by vop-cz** [comment]
> Čtvereček v bílém poli ani neznám, to by mě dost zmátlo :-))

Fakt ne? V životě jsi to neviděl? Definuje začátek/konec trasy...

**Comment by mkyral** [comment]
No tak moje manželka to taky neznala a to je docela náruživá turistka. Ale nikdy nepřemýšlela nad tím, co tahle značka vlastně znamená.

Nicméně není problém přidat do title popis dané značky (začátek modré turistické značky do xxx)

**Comment by mkyral** [comment]
Tak generátor jsem zatím nenašel. Dopracoval jsem se k tomuto: http://hiking.waymarkedtrails.org/help/rendering/osmc_legende a následně našel místo generátoru toto: https://github.com/lonvia/waymarked-trails-site/tree/master/maps/symbols/kct - sadu png obrázků. Takže zatím nic.

UPDATE: našel jsem https://github.com/bigr/map1/blob/master/generate_osmcsymbols.py - ovšem je to pod AGPL, což není kompatibilní s MIT.

**Comment by vop-cz** [comment]
> No tak moje manželka to taky neznala a to je docela náruživá turistka. 

Nejvyšší čas náruživé turistky ale i turisty touto cestou edukovat... :o) Ty značky jsem si nevymyslel, v terénu prostě jsou.

> Dopracoval jsem se k tomuto: http://hiking.waymarkedtrails.org/help/rendering/osmc_legende a následně našel místo generátoru toto: https://github.com/lonvia/waymarked-trails-site/tree/master/maps/symbols/kct - sadu png obrázků. Takže zatím nic.

A my potřebujeme za každou cenu generátor? Nestačí nám ty obrázky? Pro první použití nám to snad stačí, pro první krok máme známou konečnou sadu značek, které chceme zobrazit, ne?

**Comment by walley** [comment]
Jako naruzivy fotograf rozcestniku mohu potvrdit ze tyto znacky opravdu existuji:))

treba tady je videt (trosku z uhlu) http://map.openstreetmap.cz/img/guidepost/IMG_20140204_074751.jpg
zacatek zlute u olomouckyho hlavniho nadrazi
